dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -112,6 +115,14 @@ def self.get_cache_key(oauth_param, user)
   def register_new_user(user)
     move_oauth_params_to_cache(user)
     session["devise.user_attributes"] = user.attributes
+
+    # For some providers, signups can happen without ever having hit the sign_up page, where
+    # our tracking data is usually populated, so do it here
+    unless session[:sign_up_tracking_expiration]&.future?
+      session[:sign_up_uid] = SecureRandom.uuid.to_s
+      session[:sign_up_tracking_expiration] = 5.minutes.from_now
```

Comment:
This two- or three-step session setup happens a couple of times in this PR - might be a good candidate for extraction to something like `SignUpTracking.begin_sign_up_tracking(session)`.

Reply:
:+1: